pkg/infrastructure/kafka.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2019 Red Hat, Inc. and/or its affiliates
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package infrastructure
+
+import (
+	"fmt"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/apis/app/v1alpha1"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client"
+	"github.com/kiegroup/kogito-cloud-operator/pkg/client/kubernetes"
+	v1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"strings"
+)
+
+const (
+	strimziServerGroup = "kafka.strimzi.io"
+	// StrimziOperatorName is the default Strimzi operator name
+	StrimziOperatorName = "strimzi-cluster-operator"
+	// defaultKafkaPort default port for plain connections
+	defaultKafkaPort = 9092
+)
+
+// 9093 -> secure, 9091 -> fallback
+var kafkaFallbackPorts = []int32{9093, 9091}
+
+// IsStrimziOperatorAvailable verify if Strimzi Operator is running in the given namespace and the CRD is available
+func IsStrimziOperatorAvailable(cli *client.Client, namespace string) (available bool, err error) {
+	log.Debugf("Checking if Strimzi Operator is available in the namespace %s", namespace)
+	available = false
+	if IsStrimziAvailable(cli) {
+		log.Debugf("Strimzi CRDs available. Checking if Strimzi Operator is deployed in the namespace %s", namespace)
+		list := &v1.DeploymentList{}
+		if err = kubernetes.ResourceC(cli).ListWithNamespace(namespace, list); err != nil {
```

Comment:
Strimzi operator could be installed for the whole cluster, in which case the Deployment will be created in a different namespace.